ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -309,14 +309,26 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
         }
       }
 
-      MessageReadSet readSet = new StoreMessageReadSet(readOptions);
+      MessageReadSet readSet = new StoreMessageReadSet(readOptions, new StoreMessageReadSet.IOPHandler() {
+        @Override
+        public void onSuccess() {
+          BlobStore.this.onSuccess("GET");
+        }
+
+        @Override
+        public void onError() {
+          try {
+            BlobStore.this.onError();
+          } catch (Exception e) {
+          }
+        }
+      });
       // We ensure that the metadata list is ordered with the order of the message read set view that the
       // log provides. This ensures ordering of all messages across the log and metadata from the index.
       List<MessageInfo> messageInfoList = new ArrayList<MessageInfo>(readSet.count());
       for (int i = 0; i < readSet.count(); i++) {
         messageInfoList.add(indexMessages.get(readSet.getKeyAt(i)));
       }
-      onSuccess();
```

Comment:
Overall looks good to me.
Do you mind add your comment why we remove the onSuccess to the handler? It'll help to understand the code since the onSuccess for Get is different from all the other functions now.

Reply:
sure